raiden/raiden_service.py

```diff
@@ -658,6 +658,13 @@ def start_mediated_transfer(self, token_address, amount, identifier, target):
             if route.state == CHANNEL_STATE_OPENED
         ]
 
+        # send ping to target to make sure we can receive something back from target
+        async_result = self.protocol.send_ping(target)
+        if async_result.wait():
```

Comment:
in the non happy case this will probably block forever.

proposal:
```python
async_result.wait(timeout=.5)  # allow the ping to succeed
if async_result.ready():
  log.debug("transfer target received invitation ping")
....
```